src/plugins/data/common/search/utils.ts

```diff
@@ -11,17 +11,36 @@ import { AggTypesDependencies } from '..';
 import type { IKibanaSearchResponse } from './types';
 
 /**
+ * From https://github.com/elastic/elasticsearch/issues/55572: "When is_running is false, the query has stopped, which
+ * may happen due to ... the search failed, in which case is_partial is set to true to indicate that any results that
+ * may be included in the search response come only from a subset of the shards that the query should have hit."
  * @returns true if response had an error while executing in ES
  */
 export const isErrorResponse = (response?: IKibanaSearchResponse) => {
-  return !response || !response.rawResponse || (!response.isRunning && !!response.isPartial);
+  return (
+    !response ||
+    !response.rawResponse ||
+    (!response.isRunning &&
+      !!response.isPartial &&
+      // See https://github.com/elastic/elasticsearch/pull/97731. For CCS with ccs_minimize_roundtrips=true, isPartial
+      // is true if the search is complete but there are shard failures. In that case, the _clusters.details section
+      // will have information about those failures. This will also likely be the behavior of CCS with
```

Comment:
I am not sure that checking `_clusters.details` is enough. I think we must check _clusters.details.**clusterAlias**.failures
The new response for CCS is now including `_clusters.details` both for error and non error responses.
An example of non error `details` section is: 
```
"details": {
  "(local)": {
    "status": "successful",
    "indices": foo",
    "took": 123,
    "timed_out": false,
    "_shards": {
      ...
    }
  },
  "remote": {
    "status": "successful",
    "indices": "foo",
    "took": 234,
    "timed_out": false,
    "_shards": {
      ...
    }
  }
}
```
And an example of error `details` section:
```
"details": {
  "remote": {
      "status": "partial",
      "indices": "foo",
      "took": 123,
      "timed_out": false,
      "_shards": {
        "total": 6,
        "successful": 5,
        "skipped": 0,
        "failed": 1  
      },
      "failures": [ 
        {
          "shard": 1,
          "index": "remote:foo",
          "node": ...,
          "reason": {
            "type": "query_shard_exception",
            "reason": "failed to create query: exception message here",
            ...
          }
        }
      ]
  }
}
```